buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/CreateRuleStubsTask.kt

```diff
@@ -135,21 +140,36 @@ abstract class CreateRuleStubsTask : DefaultTask() {
     }
 }
 
-private object Templates {
-    val checksDir = Path.of("sonar-kotlin-plugin", "src", "main", "java", "org", "sonarsource", "kotlin", "checks")
-    val testsDir = Path.of("sonar-kotlin-plugin", "src", "test", "java", "org", "sonarsource", "kotlin", "checks")
-    val samplesDir = Path.of("kotlin-checks-test-sources", "src", "main", "kotlin", "checks")
-    val checkListFile =
-        Path.of("sonar-kotlin-plugin", "src", "main", "java", "org", "sonarsource", "kotlin", "plugin", "KotlinCheckList.kt")
+private interface Templates {
+    val checksDir : Path
+    val testsDir : Path
+    val samplesDir : Path
+    val checkListFile : Path
+    val checksPackage : String
+    val sampleFileExt : String
+
+    fun generateCheckClass(ruleKey: String, checkClassName: String, messageLine: String?): String
+    fun generateTestClass(testClassName: String, checkClassName: String): String
+    fun generateCheckFile(checkSampleName: String, ruleKey: String): String
+}
 
-    @OptIn(ExperimentalPathApi::class)
-    val LICENSE_HEADER by lazy {
-        Path.of("LICENSE_HEADER").readText()
-            .replace("${"$"}YEAR", Calendar.getInstance().get(Calendar.YEAR).toString())
-    }
+@OptIn(ExperimentalPathApi::class)
+private val LICENSE_HEADER by lazy {
+    Path.of("LICENSE_HEADER").readText()
+        .replace("${"$"}YEAR", Calendar.getInstance().get(Calendar.YEAR).toString())
+}
+
+private object KotlinTemplates: Templates {
+    override val checksDir = Path.of("sonar-kotlin-checks", "src", "main", "java", "org", "sonarsource", "kotlin", "checks")
+    override val testsDir = Path.of("sonar-kotlin-checks", "src", "test", "java", "org", "sonarsource", "kotlin", "checks")
+    override val samplesDir = Path.of("kotlin-checks-test-sources", "src", "main", "kotlin", "checks")
+    override val checkListFile =
+        Path.of("sonar-kotlin-plugin", "src", "main", "java", "org", "sonarsource", "kotlin", "plugin", "KotlinCheckList.kt")
+    override val checksPackage = "org.sonarsource.kotlin.checks"
+    override val sampleFileExt = "kt"
 
-    fun generateCheckClass(ruleKey: String, checkClassName: String, messageLine: String?) = LICENSE_HEADER + """
-        package org.sonarsource.kotlin.checks
+    override fun generateCheckClass(ruleKey: String, checkClassName: String, messageLine: String?) = LICENSE_HEADER + """
+        package $checksPackage
 
         import org.sonar.check.Rule
         import org.sonarsource.kotlin.api.AbstractCheck
```

Comment:
I believe this is also not the correct import anymore.

Reply:
Done.